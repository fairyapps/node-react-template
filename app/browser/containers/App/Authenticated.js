import React, { Component } from "react";
import { Switch } from "react-router-dom";
import { Route } from "react-router";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { getCurrentUser } from "../../../common/auth/selectors";
import Header from "../../components/Header";
import Profile from "../../routes/Profile";

class Authenticated extends Component {
  render() {
    const { currentUser } = this.props;

    return (
      <div>
        <Header />
        <Switch>
          <Route path="/users/:id" component={Profile} />
        </Switch>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    currentUser: getCurrentUser(state)
  };
}

export default connect(mapStateToProps)(Authenticated);
