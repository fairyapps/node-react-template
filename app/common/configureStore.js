import { createStore, applyMiddleware, compose } from "redux";
import createSagaMiddleware from "redux-saga";
import { fromJS } from "immutable";
import { routerMiddleware } from "react-router-redux";

import reducers from "./reducers";
import sagas from "./sagas";

const sagaMiddleware = createSagaMiddleware();
const composeEnhancers =
  (process.env.NODE_ENV !== "production" &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) ||
  compose;

export default function configureStore(initialState, history) {
  const store = createStore(
    reducers,
    fromJS(initialState),
    composeEnhancers(applyMiddleware(sagaMiddleware, routerMiddleware(history)))
  );

  sagaMiddleware.run(sagas);

  return store;
}
