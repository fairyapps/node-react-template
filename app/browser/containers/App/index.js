import React, { Component } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import Authenticated from "./Authenticated";
import { isLoggedIn } from "../../../common/auth/selectors";

class App extends Component {
  render() {
    const { isLoggedIn } = this.props;

    return isLoggedIn ? <Authenticated /> : null;
  }
}

function mapStateToProps(state) {
  return {
    isLoggedIn: isLoggedIn(state)
  };
}

export default connect(mapStateToProps)(App);
