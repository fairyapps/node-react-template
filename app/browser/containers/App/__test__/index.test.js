import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { fromJS } from "immutable";
import createHistory from "history/createBrowserHistory";
import configureStore from "../../../../common/configureStore";
import { ConnectedRouter } from "react-router-redux";
import App from "../index";

const history = createHistory();

test("App renders correct snapshot", () => {
  const fakeStore = state => ({
    default: () => {},
    subscribe: () => {},
    dispatch: () => {},
    getState: () => {
      return fromJS({ ...state });
    }
  });

  const store = fakeStore({
    auth: {
      isLoggedIn: true,
      currentUserId: "1"
    },
    users: {
      byId: {
        "1": {
          id: "1"
        }
      }
    }
  });

  const component = renderer.create(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <App />
      </ConnectedRouter>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
