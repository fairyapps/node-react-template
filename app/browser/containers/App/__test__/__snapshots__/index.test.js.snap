// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correct snapshot 1`] = `
<div>
  <header>
    <div
      className="sc-bdVaJa fYdnBr"
    />
  </header>
</div>
`;
